src/util/logging.cpp

```diff
@@ -30,14 +30,23 @@ QFile s_logfile;
 
 QLoggingCategory::CategoryFilter oldCategoryFilter = nullptr;
 
+/// Logging category for messages logged via the JavaScript Console API.
+constexpr char jsLoggingCategory[] = "js";
+
+/// Logging category prefix for messages logged by the controller system.
+constexpr char controllerLoggingCategoryPrefix[] = "controller.";
+
 /// Filters logging categories for the `--controller-debug` command line
 /// argument, so that debug messages are enabled for all categories in the
 /// `controller` namespace, and disabled for all other categories.
 void controllerDebugCategoryFilter(QLoggingCategory* category) {
-    // Configure controller.*.input/output category here, otherwise forward to to default filter.
-    constexpr char controllerPrefix[] = "controller.";
+    // Configure `js` or `controller.*` categories here, otherwise forward to to default filter.
     const char* categoryName = category->categoryName();
-    if (qstrncmp(categoryName, controllerPrefix, sizeof(controllerPrefix) - 1) == 0) {
+    if (qstrncmp(categoryName, jsLoggingCategory, sizeof(jsLoggingCategory)) ==
+                    0 ||
+            qstrncmp(categoryName,
+                    controllerLoggingCategoryPrefix,
+                    sizeof(controllerLoggingCategoryPrefix) - 1) == 0) {
```

Comment:
this is very hard to read. What do you think about using temporary `string_view`s instead?

```suggestion
    const std::string_view categoryNameView{category->categoryName());
    constexpr std::string_view jsLoggingCategoryView{jsLoggingCategory};
    constexpr std::string_view controllerLoggingCategoryPrefixView{controllerLoggingCategoryPrefix};
    controllerLoggingCategoryPrefixView.remove_suffix(1); // maybe just define controllerLoggingCategoryPrefix so this isn't needed
    if (categoryNameView == jsLoggingCategoryView || categoryNameView == controllerLoggingCategoryPrefixView) {
```
feel free to rename some stuff. I'm just trying to get the idea across. Performance overhead seems negligible though I can't really tell by just looking at the assembly. https://compiler-explorer.com/z/4oMo4z7ob

Reply:
It's actually just the normal strncmp, but I switched to string_view in https://github.com/mixxxdj/mixxx/pull/4612/commits/a37a12f5bf65b05b6a0b665a337d914bea126115. However, you suggestion in wrong because I don't think the `remove_suffix` will string the trailing null byte. Since we still use C++17, I doubt that the new code is much easier to read. No idea about performance.

Reply:
Its not hot code so arguing about performance is not worth the time anyway, I just wanted to mention it. IMO I find the code easier to read, even if we can't use `starts_with` just the comment alone helps because its much more expressive. As you can see from the suggestion I posted, I even misunderstood the `qstrncmp` code while your new version highlights the difference.
> However, you suggestion in wrong because I don't think the remove_suffix will string the trailing null byte.

If I'm understanding the example correctly, it should though: https://en.cppreference.com/w/cpp/string/basic_string_view/remove_suffix

Reply:
The null bytes are only part of the string_view if you initialize it like that, not if you use a normal string literal. But in any case you can't use `==` because:

> Two views are equal if both the size of lhs and rhs are equal and each character in lhs has an equivalent character in rhs at the same position.

But here the size is not equal. We used `strncmp` to compare the first n bytes and ignore the rest, which I tried to mirror  using `std::string_view::substr()` now.

Reply:
> But here the size is not equal. We used strncmp to compare the first n bytes and ignore the rest, which I tried to mirror using std::string_view::substr() now.

Yes, thats exactly what I didn't even realize in the first version so the second one makes this difference much clearer.